ui/src/message_popup/mobile_prototype/mobile_interface.jsx

```diff
@@ -0,0 +1,383 @@
+//normal imports
+import React from 'react';
+
+//other file imports
+import StudentCard from '../student_card.jsx';
+import InstructionsCard from '../instructions_card.jsx';
+import FinalSummaryCard from '../final_summary_card.jsx';
+import {TextBody, TextFooter} from './texting_interface.jsx';
+import {withStudents} from '../transformations.jsx';
+
+//material-ui imports
+import AppBar from 'material-ui/AppBar';
+import LinearProgress from 'material-ui/LinearProgress';
+import IconButton from 'material-ui/IconButton';
+import Divider from 'material-ui/Divider';
+import RaisedButton from 'material-ui/RaisedButton';
+import {fullWhite} from 'material-ui/styles/colors';
+
+//material-ui icon imports
+import MenuIcon from 'material-ui/svg-icons/navigation/menu'
+import ArrowBackIcon from 'material-ui/svg-icons/navigation/arrow-back'
+import RefreshIcon from 'material-ui/svg-icons/navigation/refresh'
+
+//
+const ONE_SECOND = 1000;
+
+export default React.createClass({
+  displayName: 'MobileInterface',
+  
+  getInitialState() {
+    const goodExamples = _.map(this.props.question.examples, (example) => {
+      return {type: 'Good', text: example};
+    });
+    const badExamples = _.map(this.props.question.nonExamples, (example) => {
+      return {type: 'Bad', text: example};
+    });
+    const allExamples = _.shuffle(_.merge(_.clone(goodExamples), badExamples));   
+    return ({
+      messages: [],
+      initialResponse: undefined,
+      revisedResponse: undefined,
+      elapsedMs: 0,
+      allExamples,
+      exampleIndex: 0
+    });
+  },
+  
+  propTypes: {
+    hasStarted: React.PropTypes.bool.isRequired,
+    onStartPressed: React.PropTypes.func.isRequired,
+    email: React.PropTypes.string.isRequired,
+    query: React.PropTypes.object.isRequired,
+    question: React.PropTypes.object,
+    questionsAnswered: React.PropTypes.number.isRequired,
+    sessionLength: React.PropTypes.number.isRequired,
+    shouldShowStudentCard: React.PropTypes.bool.isRequired,
+    onQuestionDone: React.PropTypes.func.isRequired,
+    helpType: React.PropTypes.string.isRequired,
+    limitMs: React.PropTypes.number.isRequired,
+    onLog: React.PropTypes.func.isRequired,
+    resetExperience: React.PropTypes.func.isRequired,
+    isLastQuestion: React.PropTypes.bool.isRequired,
+    responseTimes: React.PropTypes.array.isRequired
+  },
+  
+  onStart(email, sessionLength, questions, shouldShowStudentCard,shouldShowSummary, helpType){
+    this.props.onStartPressed(email, sessionLength, questions, shouldShowStudentCard,shouldShowSummary, helpType);
+  },
+  
+  addMessages(messageList){
+    var messages = _.clone(this.state.messages);
+    var initialResponse = this.state.initialResponse;
+    var revisedResponse = this.state.revisedResponse;
```

Comment:
Switch these `var`s over to `const`s where you can do indicate that they don't get mutated at all.  You can also use destructuring like this to simplify a bit:

```
const {initialResponse, revisedResponse} = this.state;
```